apps/ws/src/socket/usecases/external-services-route/external-services-route.usecase.ts

```diff
@@ -1,34 +1,52 @@
-import { Injectable } from '@nestjs/common';
+import { Injectable, Logger } from '@nestjs/common';
 
 import { MessageRepository } from '@novu/dal';
 import { ChannelTypeEnum, WebSocketEventEnum } from '@novu/shared';
-import { BullMqService } from '@novu/application-generic';
 
 import { ExternalServicesRouteCommand } from './external-services-route.command';
 import { WSGateway } from '../../ws.gateway';
 import { IUnreadCountPaginationIndication, IUnseenCountPaginationIndication } from './types';
 
+const LOG_CONTEXT = 'ExternalServicesRoute';
+
 @Injectable()
 export class ExternalServicesRoute {
-  public readonly bullMqService: BullMqService;
+  constructor(private wsGateway: WSGateway, private messageRepository: MessageRepository) {}
 
-  constructor(private wsGateway: WSGateway, private messageRepository: MessageRepository) {
-    this.bullMqService = new BullMqService();
-  }
   public async execute(command: ExternalServicesRouteCommand) {
-    if (command.event === WebSocketEventEnum.UNSEEN) {
-      await this.sendUnseenCountChange(command);
+    const isOnline = await this.connectionExist(command);
+    if (isOnline) {
+      if (command.event === WebSocketEventEnum.RECEIVED) {
+        await this.processReceivedEvent(command);
+      }
+
+      if (command.event === WebSocketEventEnum.UNSEEN) {
+        await this.sendUnseenCountChange(command);
+      }
+
+      if (command.event === WebSocketEventEnum.UNREAD) {
+        await this.sendUnreadCountChange(command);
+      }
```

Comment:
Is this still needed or it's for backwards compatible version of a worker? Since if we now only send the `RECEIVED` those will never run?

Reply:
I mentioned it here: https://github.com/novuhq/novu/pull/4152#discussion_r1323359256
Headless and Notification Center still use those events. @LetItRock confirmed.

Reply:
Very good Catch! Totally forgot about it 💪